arvo/ames/cryptography.md

```diff
@@ -0,0 +1,146 @@
+is generated with
+[`+sign:as:crub`](@/docs/arvo/reference/cryptography.md#sign-as) found in
+`sys/zuse.hoon`.
+
+Upon hearing an attestation packet, the receiving ship will generate a symmetric
+key for communications with the comet, according to the [key
+exchange](#key-exchange) protocol.
+
+The fact that the first packet exchanged between a comet and another ship must
+be an attestation packet is why comets are unable to initiate communication with
+one another, and also why comets must be the first to initiate communication
+with a non-comet. This is a technical limitation with a planned workaround.
+
+## `+acru:ames` {#crypto-core}
+
+The `+crypto-core` in `$ames-state` is an `+acru:ames` core, a
+[lead](@/docs/hoon/reference/advanced.md#dry-polymorphism-and-core-nesting-rules)
+interface core for asymmetric cryptosuites found in `sys/lull.hoon` which
+handles encryption, decryption, signing, and verifying. In practice, the only
+cryptosuite in use is [`+crub:crypto`](#crub), which implements [Suite B
+Cryptography](https://en.wikipedia.org/wiki/NSA_Suite_B_Cryptography).
+
+```hoon
+  ++  acru  $_  ^?                                      ::  asym cryptosuite
+    |%                                                  ::  opaque object
+    ++  as  ^?                                          ::  asym ops
+      |%  ++  seal  |~([a=pass b=@] *@)                 ::  encrypt to a
+          ++  sign  |~(a=@ *@)                          ::  certify as us
+          ++  sure  |~(a=@ *(unit @))                   ::  authenticate from us
+          ++  tear  |~([a=pass b=@] *(unit @))          ::  accept from a
+      --  ::as                                          ::
+    ++  de  |~([a=@ b=@] *(unit @))                     ::  symmetric de, soft
+    ++  dy  |~([a=@ b=@] *@)                            ::  symmetric de, hard
+    ++  en  |~([a=@ b=@] *@)                            ::  symmetric en
+    ++  ex  ^?                                          ::  export
+      |%  ++  fig  *@uvH                                ::  fingerprint
+          ++  pac  *@uvG                                ::  default passcode
+          ++  pub  *pass                                ::  public key
+          ++  sec  *ring                                ::  private key
+      --  ::ex                                          ::
+    ++  nu  ^?                                          ::  reconstructors
+      |%  ++  pit  |~([a=@ b=@] ^?(..nu))               ::  from [width seed]
+          ++  nol  |~(a=ring ^?(..nu))                  ::  from ring
+          ++  com  |~(a=pass ^?(..nu))                  ::  from pass
+      --  ::nu                                          ::
+    --  ::acru                                          ::
+```
+
+As the `+acru` core is merely an interface, the details on how it is utilized
+may vary according to the cryptosuite implemented in it. We summarize what each
```

Comment:
Isn't the point of the interface that you can _utilize_ it in a consistent way, but that underlying _implementation_ may vary?

Reply:
Right, I think what I had in mind was that you'd utilize different cryptosuites in different situations. Or maybe I just flubbed. Either way, you're right it is less confusing to say implemented here. https://github.com/urbit/docs/pull/1091/commits/fefe4c1695f63a65f79e7ac0a3946ff57d98553c